config/routes.rb

```diff
@@ -3,6 +3,9 @@
 
   devise_for :users
 
-  resources :events, only: %i[new create update destroy]
-  get "user_dashboard/show", as: :dashboard
+  resources :events, only: %i[create update destroy] do
+    resources :bookings, only: %i[create] do
+      patch :cancel, as: :cancel_booking, on: :member
+    end
+  end
```

Comment:
The cancel action shouldn't be nested inside the events resources. We don't need the events id to cancel a booking do we ?